src/main/java/picard/util/LiftoverUtils.java

```diff
@@ -0,0 +1,255 @@
+        }
+
+        if (builder == null){
+            return null;
+        }
+
+        builder.filters(source.getFilters());
+        builder.log10PError(source.getLog10PError());
+        builder.attributes(source.getAttributes());
+        builder.id(source.getID());
+
+        if (writeOriginalPosition) {
+            builder.attribute(LiftoverVcf.ORIGINAL_CONTIG, source.getContig());
+            builder.attribute(LiftoverVcf.ORIGINAL_START, source.getStart());
+        }
+
+        return builder.make();
+    }
+
+    protected static VariantContextBuilder liftSimpleVariantContext(VariantContext source, Interval target){
+        if (target == null || source.getReference().length() != target.length()) {
+            return null;
+        }
+
+        // Build the new variant context
+        final VariantContextBuilder builder = new VariantContextBuilder(source);
+        builder.chr(target.getContig());
+        builder.start(target.getStart());
+        builder.stop(target.getEnd());
+
+        return builder;
+    }
+
+    protected static VariantContextBuilder reverseComplementVariantContext(VariantContext source, Interval target, ReferenceSequence refSeq){
+        if (target.isPositiveStrand()){
+            throw new IllegalArgumentException("This should only be called for negative strand liftovers");
+        }
+
+        //not currently supported
+        if (source.isIndel() && !source.isBiallelic()){
+            return null;
+        }
+
+        List<Allele> origAlleles = new ArrayList<>(source.getAlleles());
+        VariantContextBuilder vcb = new VariantContextBuilder(source);
+        vcb.chr(target.getContig());
+
+        final boolean addToStart = source.isIndel() && target.getStart() > 1;
+
+        final int start = target.getStart() - (addToStart ? 1 : 0);
+        vcb.start(start);
+
+        final int stop = target.getEnd() - (addToStart ? 1 : 0);
+        vcb.stop(stop);
+
+        vcb.alleles(reverseComplementAlleles(origAlleles, target, refSeq, source.isIndel(), addToStart));
+        if (source.isIndel()){
+            leftAlignVariant(vcb, start, stop, vcb.getAlleles(), refSeq);
+        }
+
+        vcb.genotypes(fixGenotypes(source.getGenotypes(), origAlleles, vcb.getAlleles()));
+
+        return vcb;
+    }
+
+    private static List<Allele> reverseComplementAlleles(List<Allele> originalAlleles, Interval target, ReferenceSequence refSeq, boolean isBiAllelicIndel, boolean addToStart){
+        final List<Allele> alleles = new ArrayList<>();
+
+        for (final Allele oldAllele : originalAlleles) {
+            alleles.add(LiftoverUtils.reverseComplement(oldAllele, target, refSeq, isBiAllelicIndel, addToStart));
+        }
+
+        return alleles;
+    }
+
+    private static Allele reverseComplement(Allele oldAllele, Interval target, ReferenceSequence referenceSequence, boolean isBiAllelicIndel, boolean addToStart){
+
+        if (oldAllele.isSymbolic()){
```

Comment:
what about no-calls ?

Reply:
i thought this was handled upstream, but i will add a check here too

Reply:
oh and finals here too.